From bed31e0557854bb92425db1eab73022c41f2c793 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Fl=C3=B6ser?= <mgraesslin@kde.org>
Date: Sat, 5 May 2018 09:11:34 +0200
Subject: Reparse rules config prior to update

Summary:
We used to recreate the KConfig when rules needed to update. Now that it
is a KSharedConfig, which is kept, it needs to be reparsed as it changes
outside of KWin.

BUG: 393788
FIXED-IN: 5.13.0

Test Plan: Restarted session, changing rules work again

Reviewers: #kwin, #plasma

Subscribers: kwin

Tags: #kwin

Differential Revision: https://phabricator.kde.org/D12706
---
 rules.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rules.cpp b/rules.cpp
index a6060cd..32b389d 100644
--- a/rules.cpp
+++ b/rules.cpp
@@ -1064,6 +1064,8 @@ void RuleBook::load()
     deleteAll();
     if (!m_config) {
         m_config = KSharedConfig::openConfig(QStringLiteral(KWIN_NAME "rulesrc"), KConfig::NoGlobals);
+    } else {
+        m_config->reparseConfiguration();
     }
     int count = m_config->group("General").readEntry("count", 0);
     for (int i = 1;
-- 
cgit v0.11.2

